flatsurf/geometry/surface.py

```diff
@@ -562,169 +637,151 @@ class Surface_list(Surface):
         sage: # base label defaults to zero.
         sage: s.set_immutable()
         sage: TestSuite(s).run()
+
+    We surgically add a square into an infinite billiard surface::
+
+        sage: p = polygons(vertices=[(0,0),(4,0),(0,3)])
+        sage: s = similarity_surfaces.billiard(p)
+        sage: ts=s.minimal_cover(cover_type="translation").copy(relabel=True, mutable=True)
+        sage: # Explore the surface a bit
+        sage: ts.polygon(0)
+        Polygon: (0, 0), (4, 0), (0, 3)
+        sage: ts.opposite_edge(0,0)
+        (1, 2)
+        sage: ts.polygon(1)
+        Polygon: (0, 0), (0, -3), (4, 0)
+        sage: s = ts.underlying_surface()
+        sage: l=s.add_polygon(polygons.square(side=4))
+        sage: s.change_edge_gluing(0,0,l,2)
+        sage: s.change_edge_gluing(1,2,l,0)
+        sage: s.change_edge_gluing(l,1,l,3)
+        sage: print("Glued in label is "+str(l))
+        Glued in label is 2
+        sage: count = 0
+        sage: for x in ts.edge_iterator(gluings=True):
+        ....:     print(x)
+        ....:     count=count+1
+        ....:     if count>15:
+        ....:         break
+        ((0, 0), (2, 2))
+        ((0, 1), (3, 1))
+        ((0, 2), (4, 0))
+        ((2, 0), (1, 2))
+        ((2, 1), (2, 3))
+        ((2, 2), (0, 0))
+        ((2, 3), (2, 1))
+        ((3, 0), (5, 2))
+        ((3, 1), (0, 1))
+        ((3, 2), (6, 0))
+        ((4, 0), (0, 2))
+        ((4, 1), (7, 1))
+        ((4, 2), (8, 0))
+        ((1, 0), (8, 2))
+        ((1, 1), (9, 1))
+        ((1, 2), (2, 0))
+        sage: count = 0
+        sage: for l,p in ts.label_iterator(polygons=True):
+        ....:     print(str(l)+" -> "+str(p))
+        ....:     count=count+1
+        ....:     if count>5:
+        ....:         break
+        0 -> Polygon: (0, 0), (4, 0), (0, 3)
+        2 -> Polygon: (0, 0), (4, 0), (4, 4), (0, 4)
+        3 -> Polygon: (0, 0), (-72/25, -21/25), (28/25, -96/25)
+        4 -> Polygon: (0, 0), (0, 3), (-4, 0)
+        1 -> Polygon: (0, 0), (0, -3), (4, 0)
+        5 -> Polygon: (0, 0), (-28/25, 96/25), (-72/25, -21/25)
+
     """
-    ###
-    ### Brief summary of internal workings.
-    ###
-    #
-    # The Surface_list maintains a list self._p for storing polygons together
-    # with gluing data.
-    #
-    # Here self._p[label] is typically a list of two elements
-    # [polygon, gluing_list]. The gluing_list is then a list of pairs
-    # (other_label, other_edge) so that typically
-    # self.opposite_edge(label, edge) returns self._p[label][1][edge].
-    #
-    # If constructed with a surface parameter which is not None and copy=False,
-    # then Surface_list stores a reference to the provided surface as
-    # self._reference_surface. (Otherwise self._reference_surface is None).
-    #
-    def __init__(self, base_ring=None, surface = None, copy=True, mutable=None):
```

Comment:
I changed the default here. Now `copy=surface.is_mutable()`, i.e., it is more often `False`.

It seems to me that `copy=False` should have better performance in most situations in theory.

---

I also changed the default of `mutable`. The default is now `True`. When creating a surface with such low-level methods as `Surface_list` or `Surface_dict`, one usually wants to mutate the surface; otherwise, one could have kept the original surface (modulo relabeling.)

---

Both these changes do not break any existing code inside sage-flatsurf. But technically, these are breaking changes.

To me these seem to be the much more sensible defaults, but I don't mind to revert these.